engine/development/deploymentManager/src/main/scala/pl/touk/nussknacker/development/manager/DevelopmentDeploymentManagerProvider.scala

```diff
@@ -0,0 +1,137 @@
+    logger.debug(s"Starting deploying scenario: ${processVersion.processName}..")
+    val duringDeployStateStatus: ProcessState = createStateStatus(SimpleStateStatus.DuringDeploy, processVersion)
+    memory.update(processVersion.processName, duringDeployStateStatus)
+    asyncChangeState(processVersion.processName, SimpleStateStatus.Running)
+    logger.debug(s"Finished deploying scenario: ${processVersion.processName}.")
+    Future.successful(duringDeployStateStatus.deploymentId)
+  }
+
+  override def stop(name: ProcessName, savepointDir: Option[String], user: User): Future[SavepointResult] = {
+    logger.debug(s"Starting stopping scenario: $name..")
+    asyncChangeState(name, SimpleStateStatus.Finished)
+    logger.debug(s"Finished stopping scenario: $name.")
+    Future.successful(SavepointResult(""))
+  }
+
+  override def cancel(name: ProcessName, user: User): Future[Unit] = {
+    logger.debug(s"Starting canceling scenario: $name..")
+    changeState(name, SimpleStateStatus.DuringCancel)
+    asyncChangeState(name, SimpleStateStatus.Canceled)
+    logger.debug(s"Finished canceling scenario: $name.")
+    Future.unit
+  }
+
+  override def test[T](name: ProcessName, canonicalProcess: CanonicalProcess, testData: TestData, variableEncoder: Any => T): Future[TestProcess.TestResults[T]] = ???
+
+  override def findJobStatus(name: ProcessName): Future[Option[ProcessState]] =
+    Future.successful(memory.get(name))
+
+  override def savepoint(name: ProcessName, savepointDir: Option[String]): Future[SavepointResult] =
+    Future.successful(SavepointResult(""))
+
+  override def processStateDefinitionManager: ProcessStateDefinitionManager = SimpleProcessStateDefinitionManager
+
+  override def customActions: List[CustomAction] = List(
+    CustomAction("test-running", List("RUNNING")),
+    CustomAction("test-canceled", List("CANCELED")),
+    CustomAction("test-all", Nil),
+  )
+
+  override def invokeCustomAction(actionRequest: CustomActionRequest, canonicalProcess: CanonicalProcess): Future[Either[CustomActionError, CustomActionResult]] =
+    Future.successful(
+      customActions
+        .find(_.name.equals(actionRequest.name))
+        .map(_ => Right(CustomActionResult(actionRequest, s"Done${actionRequest.name}")))
+        .getOrElse(Left(CustomActionNotImplemented(actionRequest)))
+    )
+
+  override def close(): Unit = {}
+
+  private def changeState(name: ProcessName, stateStatus: StateStatus): Unit =
+    memory.get(name).foreach(processState => {
+      val newProcessState = processState.withStateStatus(stateStatus)
+      memory.update(name, newProcessState)
+      logger.debug(s"Changed scenario $name state from ${processState.status.name} to ${stateStatus.name}.")
+    })
+
+  private def asyncChangeState(name: ProcessName, stateStatus: StateStatus): Unit =
```

Comment:
Maybe actorSystem.scheduler.scheduleOnce() would be nicer? (ActorSystem is provided by DeploymentManagerProvider)

Reply:
Good point - done.